src/SingleBookingPage/NearestBooking.js

```diff
@@ -0,0 +1,202 @@
+// @flow
+
+import * as React from 'react';
+import { graphql, createFragmentContainer } from 'react-relay';
+import idx from 'idx';
+
+import BookingNotFound from './BookingNotFound';
+import OneWay from './bookingTypes/OneWay';
+import Return from './bookingTypes/Return';
+import Multicity from './bookingTypes/Multicity';
+import Contact from './bookingItem/Contact';
+import Notification from './bookingItem/Notification';
+import Header from './bookingItem/Header';
+import { URGENCY_THRESHOLD } from '../helpers/dateUtils';
+import type { NearestBooking_bookingEdges } from './__generated__/NearestBooking_bookingEdges.graphql';
+
+type Props = {|
+  bookingEdges: NearestBooking_bookingEdges,
+|};
+
+class NearestBooking extends React.Component<Props> {
+  getDate = booking => {
+    let date = 0;
+    const type = idx(booking, _ => _.type);
+
+    if (type === 'ONE_WAY') {
+      date = idx(booking, _ => _.oneWay.trip.arrival.time) || 0;
+    }
+
+    if (type === 'RETURN') {
+      date = idx(booking, _ => _.return.inbound.arrival.time) || 0;
+    }
+
+    if (type === 'MULTICITY') {
+      date = idx(booking, _ => _.multicity.end.time) || 0;
+    }
+
+    return new Date(date);
+  };
+
+  getBookingByType = booking => {
+    if (booking.type === 'ONE_WAY') {
+      return booking.oneWay;
+    }
+
+    if (booking.type === 'RETURN') {
+      return booking.return;
+    }
+
+    if (booking.type === 'MULTICITY') {
+      return booking.multicity;
+    }
+
+    return null;
+  };
+
+  renderByType = booking => {
+    const bookingByType = this.getBookingByType(booking);
+
+    if (booking.type === 'ONE_WAY') {
+      return <OneWay booking={bookingByType} />;
+    }
+
+    if (booking.type === 'RETURN') {
+      return <Return booking={bookingByType} />;
+    }
+
+    if (booking.type === 'MULTICITY') {
+      return <Multicity booking={bookingByType} />;
+    }
+
+    return null;
+  };
+
+  getNearestBooking = bookings => {
+    if (!bookings || bookings.length === 0) {
+      return null;
+    }
+
+    const bookingsWithDelta = bookings
+      .map(bookingEdge => {
+        if (!bookingEdge || !bookingEdge.node) {
+          return null;
+        }
+
+        const booking = bookingEdge.node;
+        const delta = this.getDate(booking) - new Date();
+
+        return { delta, booking: booking };
+      })
+      .filter(Boolean);
+    const nearestUpcomingBooking = bookingsWithDelta
+      .filter(({ delta }) => delta > 0)
+      .reduce((nearestBooking, bookingWithDelta) => {
+        if (!nearestBooking) {
+          return bookingWithDelta;
+        }
+
+        if (nearestBooking.delta > bookingWithDelta.delta) {
+          return bookingWithDelta;
+        }
+
+        return nearestBooking;
+      }, null);
+
+    if (nearestUpcomingBooking) {
```

Comment:
We can probably just cover from L106 to L111 with `return nearestUpcomingBooking` or if `null` is specifically desired `return nearestUpcomingBooking || null`

Reply:
flow unfortunately understand only `.filter(Boolean)`